libpromises/evalfunction.c

```diff
@@ -8722,6 +8722,28 @@ static FnCallResult FnCallCFEngineCallers(EvalContext *ctx, ARG_UNUSED const Pol
     return FnReturnContainerNoCopy(callers);
 }
 
+static FnCallResult FnCallString(EvalContext *ctx,
+                                 const Policy *policy,
+                                 const FnCall *fp,
+                                 const Rlist *finalargs)
+{
+    // Parser doesnt allow this to be true since STRING_ARGS has one argument
+    if(!finalargs) {
+        return FnFailure();
+    }
+
+    char* ret;
+    switch(finalargs->val.type) {
+        case RVAL_TYPE_SCALAR:
+            ret = RvalToString(finalargs->val);
+            break;
+        default:
+            ProgrammingError("Shouldnt reach this TODO");
+    }
```

Comment:
Switch is overkill but i wrote it expecting to deal with different finalargs types. If the functions arguments are `CF_DATA_TYPE_STRING` can i expect that cfengine wont pass anything else?